components/hab/src/main.rs

```diff
@@ -1621,6 +1632,11 @@ fn get_group_from_input(m: &ArgMatches) -> Option<String> {
     m.value_of("GROUP").map(ToString::to_string)
 }
 
+fn get_install_hook_mode_from_input(m: &ArgMatches) -> Option<InstallHookMode> {
+    m.value_of("INSTALL_HOOK_MODE")
+        .and_then(|f| InstallHookMode::from_str(f).ok())
+}
```

Comment:
I think `map` (closure `T -> U`) is more ergonomic than `and_then` (closure `T -> Option(U)`) since it obviates the `ok()`.

However, If you take my suggestion above and convert this to do the default handling, here's what I'd suggest:
```rust
fn get_install_hook_mode(m: &ArgMatches) -> InstallHookMode {
    match m.value_of("INSTALL_HOOK_MODE") {
        // unwrap ok because of valid_install_hook_mode has already run
        Some(s) => InstallHookMode::from_str(s).unwrap(),
        None => InstallHookMode::default(),
    }
}
```